### ISSUE TITLE
Bug report: DataFrame resample fails for some timezones

### ISSUE THREAD

**Issue by dfonnegra** [dask/dask#5746]
When a resample bin falls in a DST change datetime, the resample fails with NonExistentTimeError. This was a bug solved in pandas version 0.24.0 and what they do when a datetime doesnt exist by dst change is making a shift to the closest existing datetime.

To replicate the error copy and paste this:
```
import pandas as pd
import dask.dataframe as ddf
df = pd.DataFrame([[1], [2], [3]], index=[pd.Timestamp("2016-10-15 00:00:00"), pd.Timestamp("2016-10-16 10:00:00"), pd.Timestamp("2016-10-17 00:00:00")])
df.index = df.index.tz_localize('America/Sao_Paulo')
ddf2 = ddf.from_pandas(df, npartitions=1)
ddf2.resample('1D').mean().compute()
```

I suggest changing the following instruction at dask/dataframe/tseries/resample.py in the _resample_series function:
```
pd.date_range(start, end, freq=rule, closed=reindex_closed, name=out.index.name)
```
by the following one:
```
pd.date_range(start.tz_localize(None), end.tz_localize(None), freq=rule, closed=reindex_closed, name=out.index.name).tz_localize(start.tz, nonexistent="shift_forward")
```

**Comment by mrocklin** [comment]
Thank you for the excellent bug report and reproducible example @dfonnegra .  I apologise for the long delay in response. 

It sounds like you have a solid idea for a fix.  Would you be willing to submit a PR?

Also, cc'ing resident Pandas expert @TomAugspurger just for his awareness.

**Comment by TomAugspurger** [comment]
@dfonnegra are you interested in working on this? Your suggested fix sounds good.

**Comment by dfonnegra** [comment]
@TomAugspurger Yes Tom! Sorry, the begining of the year has been a little hard, but I'll submit a PR soon...

**Comment by TomAugspurger** [comment]
Great, thanks! No rush of course :)

**Comment by dfonnegra** [comment]
Solved in Pull Request #5807

**Comment by TomAugspurger** [comment]
Closed by #5807
